src/compiler/compiler-core/src/parser.ts

```diff
@@ -460,24 +460,35 @@ export class ProjectParser extends EventEmitter implements Parser {
   }
 
   private locateSources (sourceFile: SourceFile, sourceMap: Map<string, string>) {
-    if (!sourceFile.compilerNode.resolvedModules) {
-      return;
-    }
+    const literalsReferencingOtherSourceFiles = sourceFile.getLiteralsReferencingOtherSourceFiles();
+    const referencedSourceFiles = sourceFile.getReferencedSourceFiles();
+
+    for (let index = 0; index < referencedSourceFiles.length; index++) {
+      // It's safe to use the same index for both arrays, internally they are both generated from the same `Map`
```

Comment:
Just calling this out as surface-area for a possible future bug, since we're somewhat-breaking encapsulation by assuming this is safe. (Perhaps a future version of ts-morph will change its internal implementation, quietly breaking our logic here.)

I don't think this merits adjusting the PR, as I'm not sure if there's a better way forward.  Maybe this counts as a 'nit' :)

Reply:
I think this is a good point, I will find another way around this.

Reply:
fixed, thanks!